src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+
+            if (isLibraryPage) {
+                bodyClassList.add("libraryDocument");
+                bodyClassList.remove("dashboardDocument");
+                bodyClassList.remove("hideMainDrawer");
+
+                if (navDrawerInstance) {
+                    navDrawerInstance.setEdgeSwipeEnabled(true);
+                }
+            } else {
+                if (isDashboardPage) {
+                    bodyClassList.remove("libraryDocument");
+                    bodyClassList.add("dashboardDocument");
+                    bodyClassList.remove("hideMainDrawer");
+
+                    if (navDrawerInstance) {
+                        navDrawerInstance.setEdgeSwipeEnabled(true);
+                    }
+                } else {
+                    bodyClassList.remove("libraryDocument");
+                    bodyClassList.remove("dashboardDocument");
+                    bodyClassList.add("hideMainDrawer");
+
+                    if (navDrawerInstance) {
+                        navDrawerInstance.setEdgeSwipeEnabled(false);
+                    }
+                }
+            }
+        }
+
+        if (requiresUserRefresh) {
+            connectionManager.user(getCurrentApiClient()).then(updateUserInHeader);
         }
-        requiresUserRefresh && connectionManager.user(getCurrentApiClient()).then(updateUserInHeader)
     }
 
     function updateTitle(page) {
         var title = page.getAttribute("data-title");
-        title ? LibraryMenu.setTitle(title) : page.classList.contains("standalonePage") && LibraryMenu.setDefaultTitle()
+
+        if (title) {
+            LibraryMenu.setTitle(title);
+        } else {
+            if (page.classList.contains("standalonePage")) {
+                LibraryMenu.setDefaultTitle();
+            }
+        }
     }
 
     function updateBackButton(page) {
-        headerBackButton || (headerBackButton = document.querySelector(".headerBackButton")), headerBackButton && ("false" !== page.getAttribute("data-backbutton") && appRouter.canGoBack() ? headerBackButton.classList.remove("hide") : headerBackButton.classList.add("hide"))
+        var isDashboardPage = page.classList.contains("type-interior");
+        
+        if (!headerBackButton) {
+            headerBackButton = document.querySelector(".headerBackButton");
+        }
+
+        if (headerBackButton) {
+            if ("false" !== !isDashboardPage && page.getAttribute("data-backbutton") && appRouter.canGoBack()) {
```

Comment:
Hmm. Isn't `isDashboardPage` a boolean? `"false"` is a string, so they will never be equal.
```suggestion
            if (!isDashboardPage && page.getAttribute("data-backbutton") && appRouter.canGoBack()) {
```

Reply:
And yet, this is the single bit of code I had to write to make the back button not appear on the dashboard.

Think it has to do with the getter. If its there, it returns true. Its used further up in the original code, hence why I used this to fix it up.

I can try the alternative you suggest, but changing that line has resulted in really stupid outcomes like total UI failure.

Reply:
All I can say is, that `"false" !== !false` will always be false. The triple equality check matches the type as well as the value and a string and a boolean are not the same type.

Reply:
well... `"false" !== page.getAttribute("data-backbutton") && appRouter.canGoBack()` is what it was before i changed it, and removing the `"false !=="` broke the function entirely.

Adding in `!isDashboardPage` at the beginning not only keeps it going for non-dashboard pages, it exempts the two I wanted to from having the back button shown.

Reply:
This fix cannot be correct, I am with @cvium here. There should be some proper solution.